TeamZeroServer/src/server/DbConnection.java

```diff
@@ -180,6 +180,40 @@ public Client authenticateUser(String userName, String password) {
 		}
 		return allClients;
 	} 
+	
+	/**
+	 * Gets a list of users from the database that match the query.
+	 * 
+	 * @return an arraylist of Client objects
+	 */
+	public ArrayList<Client> getSearchedUsers(String query) {
+		ArrayList<Client> searchedClients = new ArrayList<Client>();
+		Connection conn = this.connect();
+		try {
+			PreparedStatement ps = conn.prepareStatement("SELECT * FROM USERS WHERE lower(username) LIKE ?;");
```

Comment:
The query's parameter needs to be explicitly made lowercase too (i.e. lower(username) LIKE lower(?) ) to be case insensitive